tests/Microsoft.Identity.Web.Test/AuthorityHelpersTests.cs

```diff
@@ -118,5 +118,205 @@ public void EnsureAuthorityIsV2(string initialAuthority, string expectedAuthorit
+        public void AddAuthorityQueryToOptions(AuthorityHelpersTheoryData theoryData)
+        {
+            // arrange
+            MicrosoftIdentityOptions options = new MicrosoftIdentityOptions
+            {
+                Authority = theoryData.Authority,
+            };
+
+            // act
+            AuthorityHelpers.AddAuthorityQueryToOptions(options);
+
+            // assert
+            Assert.NotNull(options.ExtraQueryParameters);
+            Assert.Equal(theoryData.ExpectedExtraQueryParameters.Count, options.ExtraQueryParameters.Count);
+            foreach (var key in theoryData.ExpectedExtraQueryParameters.Keys)
+            {
+                Assert.True(options.ExtraQueryParameters.ContainsKey(key));
+                Assert.Equal(theoryData.ExpectedExtraQueryParameters[key], options.ExtraQueryParameters[key]);
+            }
+        }
+
+        public static TheoryData<AuthorityHelpersTheoryData> AddAuthorityQueryToOptionsTheoryData()
+        {
+            var singleQuery = "?key1=value1";
+            var multipleQueries = "?key1=value1&key2=value2";
+            var emptyQuery = "?";
+            var queryNoValue = "?key1";
+
+            var singleExpectedExtraQueryParams = new Dictionary<string, string>
+            {
+                { "key1", "value1" }
+            };
+            
+            var multipleExpectedExtraQueryParams = new Dictionary<string, string>
+            {
+                { "key1", "value1" },
+                { "key2", "value2" }
+            };
+
+            var emptyExpectedExtraQueryParams = new Dictionary<string, string>();
+
+            var theoryData = new TheoryData<AuthorityHelpersTheoryData>
+            {
+                new("AuthorityCommonTenant_SingleQuery")
+                {
+                    Authority = TestConstants.AuthorityCommonTenant + singleQuery,
+                    ExpectedExtraQueryParameters = singleExpectedExtraQueryParams
+                },
+                new("AuthorityCommonTenant_MultipleQueries")
+                {
+                    Authority = TestConstants.AuthorityCommonTenant + multipleQueries,
+                    ExpectedExtraQueryParameters = multipleExpectedExtraQueryParams
+                },
+                new("AuthorityCommonTenant_EmptyQuery")
+                {
+                    Authority = TestConstants.AuthorityCommonTenant + emptyQuery,
+                    ExpectedExtraQueryParameters = emptyExpectedExtraQueryParams
+                },
+                new("AuthorityCommonTenant_QueryNoValue")
+                {
+                    Authority = TestConstants.AuthorityCommonTenant + queryNoValue,
+                    ExpectedExtraQueryParameters = emptyExpectedExtraQueryParams
+                },
+                new("AuthorityOrganizationsUSTenant_SingleQuery")
+                {
+                    Authority = TestConstants.AuthorityOrganizationsUSTenant + singleQuery,
```

Comment:
nit: the US authority is not needed as the format is the same as the common.

Reply:
Done!